core/trino-main/src/main/java/io/trino/sql/planner/planprinter/HashingAnonymizer.java

```diff
@@ -0,0 +1,355 @@
+import io.trino.execution.TableInfo;
+import io.trino.metadata.IndexHandle;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.metadata.TableExecuteHandle;
+import io.trino.metadata.TableHandle;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ConnectorPartitioningHandle;
+import io.trino.spi.type.Type;
+import io.trino.sql.planner.PartitioningHandle;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.SystemPartitioningHandle;
+import io.trino.sql.tree.BinaryLiteral;
+import io.trino.sql.tree.BooleanLiteral;
+import io.trino.sql.tree.CharLiteral;
+import io.trino.sql.tree.DecimalLiteral;
+import io.trino.sql.tree.DoubleLiteral;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.ExpressionRewriter;
+import io.trino.sql.tree.ExpressionTreeRewriter;
+import io.trino.sql.tree.GenericLiteral;
+import io.trino.sql.tree.IntervalLiteral;
+import io.trino.sql.tree.Literal;
+import io.trino.sql.tree.LongLiteral;
+import io.trino.sql.tree.NullLiteral;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.sql.tree.SymbolReference;
+import io.trino.sql.tree.TimeLiteral;
+import io.trino.sql.tree.TimestampLiteral;
+
+import java.util.Locale;
+import java.util.Map;
+
+import static com.google.common.hash.Hashing.murmur3_128;
+import static io.trino.sql.planner.Partitioning.ArgumentBinding;
+import static io.trino.sql.planner.plan.StatisticsWriterNode.WriteStatisticsHandle;
+import static io.trino.sql.planner.plan.StatisticsWriterNode.WriteStatisticsTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.CreateTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.DeleteTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.InsertTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.RefreshMaterializedViewTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.TableExecuteTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.UpdateTarget;
+import static io.trino.sql.planner.plan.TableWriterNode.WriterTarget;
+import static io.trino.sql.tree.ExpressionTreeRewriter.rewriteWith;
+import static java.lang.Math.log10;
+import static java.lang.Math.pow;
+import static java.lang.Math.round;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+/**
+ * An implementation of {@link Anonymizer} which uses murmur3 hashing to anonymize the plan
+ */
+public class HashingAnonymizer
+        implements Anonymizer
+{
+    private enum ObjectType
+    {
+        CATALOG,
+        SCHEMA,
+        TABLE,
+        COLUMN,
+        SYMBOL,
+        LITERAL
+    }
+
+    private final String environment;
+
+    public HashingAnonymizer(String environment)
+    {
+        this.environment = requireNonNull(environment, "environment is null");
+    }
+
+    @Override
+    public String anonymize(Type type, String value)
+    {
+        return type.getDisplayName() + "_value";
```

Comment:
The type name can contain spaces and parenthesis. For a type such as `timestamp(3) with time zone`, the result of this method will be `timestamp(3) with time zone_value`.